src/main/java/Jason.java

```diff
@@ -0,0 +1,208 @@
+            System.out.println(list[taskNumber]);
+        } catch (NumberFormatException e) {
+            throw new JasonException("Please enter a valid task number.");
+        }
+    }
+
+    private static void unmarkTask(String[] input) throws JasonException {
+        try {
+            int taskNumber = Integer.parseInt(input[1]) - 1;
+            if (taskNumber < 0 || taskNumber >= num) {
+                throw new JasonException("That task number is not valid. The list only has " + num + " task(s) currently");
+            }
+
+            list[taskNumber].markAsNotDone();
+            System.out.println("OK, I've marked this task as not done yet:");
+            System.out.println(list[taskNumber]);
+        } catch (NumberFormatException e) {
+            throw new JasonException("Please enter a valid task number.");
+        }
+    }
+
+
+    private static void showTaskNumber() {
+        if (num == 1) {
+            System.out.println("Now you have 1 task in the list");
+        } else {
+            System.out.println("Now you have " + num + " tasks in the list");
+        }
+
+    }
+
+    private static void showList() throws JasonException {
+        if (num == 0) {
+            throw new JasonException("The task list is currently empty. Add some tasks!");
+        }
+
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < num; i++) {
+            System.out.println(i + 1 + ":" + list[i]);
+        }
+    }
+
+    private static void todoTasks(String task) throws JasonException {
+        String taskDescription = task.substring(5).trim(); // trim() removes leading and trailing spaces
+        if (taskDescription.isEmpty()) {
+            throw new JasonException("The description of a todo cannot be empty.");
+        }
+        list[num] = new Todo(taskDescription);
+        System.out.println("Got it. I've added this task:");
+        System.out.println(list[num].toString());
+        num++;
+        showTaskNumber();
+    }
+
+    private static void deadlineTasks(String task) throws JasonException {
+        if (!task.contains("/by")) {
+            throw new JasonException("Invalid task. Please type in the format: description /by (date/time)");
+        }
+        String[] parts = task.split("/by", 2);
+        String taskDescription = parts[0].substring(9).trim();
+        String taskDeadlineBy = parts[1].trim();
+
+        if (taskDescription.isEmpty() || taskDeadlineBy.isEmpty()) {
+            throw new JasonException("Invalid task. Description and date/time cannot be empty.");
+        }
+        list[num] = new Deadline(taskDescription, taskDeadlineBy);
+        System.out.println("Got it. I've added this task:");
+        System.out.println(list[num].toString());
+        num++;
+        showTaskNumber();
+    }
+
+
+
+    private static void eventTasks(String task) throws JasonException {
+        if (!task.contains("/from") || !task.contains("/to")) {
```

Comment:
avoid the usage of magic literals by using static final variables. 
https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers